### PR TITLE
Fix SR failing pending requests

### PR DESCRIPTION
**Pull request by Haarolean**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

**What changes did you make?** (Give an overview)
- Fixed schema registry not returning a response
- Got rid of unnecessary (by-design) schema duplicate check which made SR internal cache go crazy thus returning 404 on DELETE of existing schemas.

**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [x] Manually (please, describe, if necessary)
- [ ] Unit checks
- [ ] Integration checks
- [x] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**
telegram-cloud-photo-size-2-5427231790780758236-y

### CODE REVIEW

**Comment by Haarolean** [comment]
@jonasgeiregat FYI

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/exception/ErrorCode.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
   KSQL_API_ERROR(5001, HttpStatus.INTERNAL_SERVER_ERROR),
   BINDING_FAIL(4001, HttpStatus.BAD_REQUEST),
   NOT_FOUND(404, HttpStatus.NOT_FOUND),
-  INVALID_ENTITY_STATE(4001, HttpStatus.BAD_REQUEST),
+  INVALID_ENTITY_STATE(4016, HttpStatus.BAD_REQUEST),
```
Comment (Haarolean) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/exception/ErrorCode.java:14]: There's already one with this code :)

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/SchemaRegistryService.java [hunk 2]
```diff
@@ -265,11 +252,15 @@ private SchemaSubjectDTO withSchemaType(SchemaSubjectDTO s) {
 
   public Mono<CompatibilityLevelDTO> getSchemaCompatibilityLevel(KafkaCluster cluster,
                                                                  String schemaName) {
-    String configEndpoint = Objects.isNull(schemaName) ? "/config" : "/config/{schemaName}";
+    String globalConfig = Objects.isNull(schemaName) ? "/config" : "/config/{schemaName}";
+    final var values = new LinkedMultiValueMap<String, String>();
+    values.add("defaultToGlobal", "true");
```
Comment (Haarolean) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/SchemaRegistryService.java:257]: [Get](https://docs.confluent.io/platform/current/schema-registry/develop/using.html#show-compatibility-requirements-in-effect-for-a-subject) global compatibility level if nothing is specified for the subject

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/SchemaRegistryService.java [hunk 3]
```diff
@@ -313,29 +305,50 @@ private void setBasicAuthIfEnabled(InternalSchemaRegistry schemaRegistry, HttpHe
...
-          "You specified password but do not specified username");
+          "You specified password but did not specify username");
     }
   }
 
+  private boolean isUnrecognizedFieldSchemaTypeMessage(String errorMessage) {
+    return errorMessage.contains(UNRECOGNIZED_FIELD_SCHEMA_TYPE);
+  }
+
+  private WebClient.RequestBodySpec configuredWebClient(KafkaCluster cluster, HttpMethod method, String uri) {
```
Comment (Haarolean) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/SchemaRegistryService.java:319]: The rest is refactoring required for passing uri variables / query params.

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/SchemaRegistryService.java [hunk 4]
```diff
@@ -183,54 +192,31 @@ private SchemaSubjectDTO withSchemaType(SchemaSubjectDTO s) {
...
                 .flatMap(x -> Mono.error(isUnrecognizedFieldSchemaTypeMessage(x.getMessage())
-                        ? new SchemaTypeIsNotSupportedException()
-                        : new UnprocessableEntityException(x.getMessage()))))
+                    ? new SchemaTypeIsNotSupportedException()
+                    : new UnprocessableEntityException(x.getMessage()))))
         .bodyToMono(SubjectIdResponse.class);
   }
 
-  @NotNull
-  private Mono<SchemaSubjectDTO> checkSchemaOnDuplicate(String subject,
```
Comment (Haarolean) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/SchemaRegistryService.java:213]: There's no need for this. Moreover, this caused SR cache to save schema id which triggered 404 on delete in the future.

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/exception/ErrorCode.java (hunk 1 above)
Comment (iliax) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/exception/ErrorCode.java:14]: can you also fix 4015 ?)

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/SchemasController.java [hunk 5]
```diff
@@ -118,23 +117,23 @@ protected KafkaCluster getCluster(String clusterName) {
...
-              var totalPages = (filteredSubjects.size() / pageSize)
-                      + (filteredSubjects.size() % pageSize == 0 ? 0 : 1);
-              List<String> subjectsToRender = filteredSubjects.stream()
-                      .skip(subjectToSkip)
-                      .limit(pageSize)
-                      .collect(Collectors.toList());
-              return schemaRegistryService.getAllLatestVersionSchemas(getCluster(clusterName), subjectsToRender)
-                      .map(a -> new SchemaSubjectsResponseDTO().pageCount(totalPages).schemas(a));
-            }).map(ResponseEntity::ok);
+        .getAllSubjectNames(getCluster(clusterName))
```
Comment (iliax) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/SchemasController.java:120]: we need this reformatting?

**Review by iliax — APPROVED** [review]

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/SchemasController.java (hunk 5 above)
Comment (Haarolean) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/SchemasController.java:120]: Yea, next method call starts one symbol after the previous statement, not a symbol after `return`, that's weird. Also no clue why checkstyle hasn't failed before.

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/exception/ErrorCode.java (hunk 1 above)
Comment (Haarolean) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/exception/ErrorCode.java]: Yea, done

**Pull request merged**